### DISCUSSION TITLE
False Positive for Atlassian API Token in a binary file

### DISCUSSION THREAD

**Discussion by jesseditson** [community/community#127297] [Code Security]
### Select Topic Area

Bug

### Body

Hey there! I have a template repo that includes some binary files for compiling/running the code in the repo. This template repo will be cloned quite a lot, and every time I clone it I get a false positive. This will freak out my users so it'd be nice to get either the template repo excluded from this check or otherwise create a solution for skipping binary-like files. We could try denylisting this specific token but I'm concerned that when I update the binary in the template repo, it'll trigger this check again with a different but similar sequence, which is common in these binaries.
If it helps, these binaries are generated by cargo.

Example:
https://github.com/jesseditson/rigsketball-signup/security/secret-scanning/1

**Comment by aashah** [comment]
Hey @jesseditson , we're not able to see the secret scanning alert for privacy/security reasons. If you're comfortable, and confident it is a false-positive, could you share the value here?

**Reply by jesseditson** [reply]
Yeah no problem!

3fmt17hc8b47544cacbd630E

**Reply by jesseditson** [reply]
There are no secrets embedded in the binary, and I don't use altassian products, so I'm certain it's a false positive.

**Reply by aashah** [reply]
Oof, this one is pretty unfortunate. Atlassian keys in general have checksum information embedded in them, and this token is passing them. 😔

**Reply by aashah** [reply]
Could you share some of the text surrounding that token? Perhaps with some of the context around it, we can tighten our matching.

**Reply by jesseditson** [reply]
oops sorry just getting back to this. Yup here's a broader paste:

```
onzero..NonZeroU64$u20$as$u20$core..fmt..Debug$GT$3fmt17hc8b47544cacbd630E^@{impl#83}^@_ZN4core3fmt3num52_$LT$impl$u20$core..fmt..Debug$u20$for$u20$isize$GT$3fmt17haa6d4bcaed906815E^@{impl#80}^@_ZN4core3f
```

Another possible solution (although perhaps it's a goal of this feature to check inside binaries) would be to check the file for an ELF header before scanning files, as it seems plausible that additional checksum-passing "random" strings could be fairly easily generated the same way I hit it.
